staging/src/k8s.io/kubectl/pkg/cmd/cmd.go

```diff
@@ -323,6 +327,12 @@ func NewKubectlCommand(o KubectlOptions) *cobra.Command {
 	if kubeConfigFlags == nil {
 		kubeConfigFlags = defaultConfigFlags
 	}
+	if os.Getenv("ENABLE_KUBERC") != "" {
+		klog.Infoln("kuberc is enabled!")
+		if o.KubercHandler != nil {
+			o.KubercHandler.InjectOverridesRoot(kubeConfigFlags, o.Arguments)
```

Comment:
I think this should be below of `kubeConfigFlags.AddFlags(flags)` not above(during my local tests, if it is above the `AddFlags`, defaulting did not work.)

Have you tried only injecting defaults in kuberc file and seen what cobra does when user explicitly passes flag after `AddFlags` executed?

I mean;

```go
kubeConfigFlags.AddFlags(flags)
if cmdutil.KubeRc.IsEnabled() && o.KubercHandler != nil {
o.KubercHandler.InjectOverridesRoot(kubeconfigflags) // kubeconfigflags are set with kuberc settings via flag.Set() function 
}
```

Reply:
I believe that causes a break in the setting of the flags for the subcommands (minus config) but I need to double check.